### PR TITLE
Cleanup for 2.0 release

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

File: code/analytics/build.gradle [hunk 1]
```diff
@@ -67,10 +67,8 @@ dependencies {
     testImplementation fileTree(include: ['*.aar'], dir: 'libs')
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test:runner:1.5.2'
-    androidTestImplementation 'com.adobe.marketing.mobile:core:2.0.0-SNAPSHOT'
-    androidTestImplementation('com.adobe.marketing.mobile:identity:2.0.0-SNAPSHOT'){
-        transitive = false
-    }
+    androidTestImplementation 'com.adobe.marketing.mobile:core:2.0.0'
+    androidTestImplementation 'com.adobe.marketing.mobile:identity:2.0.0'
```
Comment (kevinlind) [inline code/analytics/build.gradle:71]: Use "2.+" here for the test dependency.

File: code/build.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ allprojects {
         mavenCentral()
         google()
         maven { url "https://jitpack.io"  }
-        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+//        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
```
Comment (kevinlind) [inline code/build.gradle:24]: Shouldn't the jitpack repo be removed while the Sonatype repo be kept?

File: code/testapp/build.gradle [hunk 3]
```diff
@@ -46,21 +46,17 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.0'
 
     implementation project(":analytics")
-
-    implementation 'com.adobe.marketing.mobile:core:2.0.0-SNAPSHOT'
-//    implementation('com.adobe.marketing.mobile:analytics:2.0.0-SNAPSHOT'){
+    //    implementation('com.adobe.marketing.mobile:analytics:2.0.0-SNAPSHOT'){
 //        transitive = false
 //    }
```
Comment (kevinlind) [inline code/testapp/build.gradle:51]: Remove these commented out lines.

File: code/testapp/build.gradle [hunk 4]
```diff
@@ -46,21 +46,17 @@ dependencies {
...
-        transitive = false
-    }
-    implementation('com.github.adobe:aepsdk-assurance-android:2.0.0-alpha.1'){
-        transitive = false
-    }
 
+    implementation 'com.adobe.marketing.mobile:core:2.0.0'
+    implementation 'com.adobe.marketing.mobile:lifecycle:2.0.0'
+    implementation 'com.adobe.marketing.mobile:identity:2.0.0'
+    implementation 'com.adobe.marketing.mobile:assurance:2.0.0'
```
Comment (kevinlind) [inline code/testapp/build.gradle:56]: Can you set each of these to "2.+"?

File: Documentation/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Analytics extension has a dependency on [AEP Core SDK](https://github.com/adobe/aepsdk-core-android#readme) which must be installed to use the extension.
+
+## Add Analytics extension to your app
+
+1. Installation via [Maven](https://maven.apache.org/) & [Gradle](https://gradle.org/) is the easiest and recommended way to get the AEP SDK into your Android app. Add a dependency on Analytics and Core to your mobile application. To ensure consistent builds, it is best to explicitly specify the dependency version and update them manually.
+
+   ```
+   implementation 'com.adobe.marketing.mobile:core:2.+'
+   implementation 'com.adobe.marketing.mobile:analytics:2.+'
+   ```
```
Comment (kevinlind) [inline Documentation/getting-started.md:14]: Add a like to our latest release page, similar to the example for Edge, https://github.com/adobe/aepsdk-edge-android/blob/dev-v2.0.0/Documentation/getting-started.md#add-the-edge-network-extension-to-your-app

File: Documentation/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+Analytics extension has a dependency on [AEP Core SDK](https://github.com/adobe/aepsdk-core-android#readme) which must be installed to use the extension.
+
+## Add Analytics extension to your app
+
+1. Installation via [Maven](https://maven.apache.org/) & [Gradle](https://gradle.org/) is the easiest and recommended way to get the AEP SDK into your Android app. Add a dependency on Analytics and Core to your mobile application. To ensure consistent builds, it is best to explicitly specify the dependency version and update them manually.
+
+   ```
+   implementation 'com.adobe.marketing.mobile:core:2.+'
+   implementation 'com.adobe.marketing.mobile:analytics:2.+'
```
Comment (kevinlind) [inline Documentation/getting-started.md:13]: Should Identity be included here as well?

File: Documentation/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+   ```java
+   public class MainApp extends Application {
+        private static final String APP_ID = "YOUR_APP_ID";
+
+        @Override
+        public void onCreate() {
+            super.onCreate();
+
+            MobileCore.setApplication(this);
+            MobileCore.setLogLevel(LoggingMode.VERBOSE);
```
Comment (kevinlind) [inline Documentation/getting-started.md:45]: Can you remove this line which sets the log level?  It's an optional step and we don't want it to seem like we recommend setting VERBOSE in production.

File: Documentation/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+   ### Kotlin
+
+   ```kotlin
+   class MyApp : Application() {
+
+       override fun onCreate() {
+           super.onCreate()
+           MobileCore.setApplication(this)
+           MobileCore.setLogLevel(LoggingMode.VERBOSE)
```
Comment (kevinlind) [inline Documentation/getting-started.md:65]: Same here, remove setting the log level.

File: code/testapp/build.gradle (hunk 4 above)
Comment (yangyansong-adbe) [reply code/testapp/build.gradle:56]: I don't think we should use `+`, the reason has been posted above.

File: code/build.gradle [hunk 9]
```diff
@@ -20,8 +20,8 @@ allprojects {
     repositories {
         mavenCentral()
         google()
-        maven { url "https://jitpack.io"  }
-        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
+//        maven { url "https://jitpack.io"  }
+//        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
```
Comment (kevinlind) [inline code/build.gradle:24]: Here, can you just remove the line for jitpack? It's my understanding that we are not using it anymore after the initial release.
You can keep Sonatype uncommented as we'll still want to pull snapshots during development.

File: Documentation/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+        @Override
+        public void onCreate() {
+            super.onCreate();
+
+            MobileCore.setApplication(this);
+            MobileCore.configureWithAppID(APP_ID);
+
+            List<Class<? extends Extension>> extensions = Arrays.asList(
+                    Analytics.EXTENSION, Identity.EXTENSION, ...);
```
Comment (kevinlind) [inline Documentation/getting-started.md:54]: Can you remove the "..." in the Array. I think these examples should be working code. Including Analytics and Identity is the minimum to have Analytics working, so I feel this is a good example to get people up and running with the extension.

File: Documentation/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+   ```kotlin
+   class MyApp : Application() {
+
+       override fun onCreate() {
+           super.onCreate()
+           MobileCore.setApplication(this)
+           MobileCore.configureWithAppID("YOUR_APP_ID")
+
+           val extensions = listOf(Analytics.EXTENSION, Identity.EXTENSION, ...)
```
Comment (kevinlind) [inline Documentation/getting-started.md:72]: Same as above, remove the "..." from the list.

File: code/analytics/build.gradle (hunk 1 above)
Comment (emdobrin) [reply code/analytics/build.gradle:70]: @yangyansong-adbe for test app/tests we will always want to use with latest available source code and catch issues early. I am thinking we can use 2.+ for now for these test targets and we logged a task to look into ways to update to latest available version and lock in the version similar with the ios podlock

File: Documentation/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
+# Getting Started with Analytics SDK
+
+## Before starting
+
+Analytics extension has a dependency on [AEP Core SDK and AEP Identity SDK](https://github.com/adobe/aepsdk-core-android#readme) which must be installed to use the extension.
```
Comment (emdobrin) [inline Documentation/getting-started.md:5]: on the Mobile Core and Identity extensions

File: Documentation/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+## Add Analytics extension to your app
+
+1. Installation via [Maven](https://maven.apache.org/) & [Gradle](https://gradle.org/) is the easiest and recommended way to get the AEP SDK into your Android app. Add a dependency on Analytics and Core to your mobile application. To ensure consistent builds, it is best to explicitly specify the dependency version and update them manually.
```
Comment (emdobrin) [inline Documentation/getting-started.md:9]: way for getting the Mobile SDK. Add a dependency on Analytics and Mobile Core

**Comment by cacheung** [comment]
Please update the license to 2023
https://github.com/adobe/aepsdk-analytics-android/blob/dev/LICENSE#L189

File: Documentation/getting-started.md (hunk 10 above)
Comment (emdobrin) [inline Documentation/getting-started.md:54]: let's remove ... for a complete sample - same for kotlin

File: Documentation/getting-started.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+   import com.adobe.marketing.mobile.Analytics
+   ```
+
+3. Import the Analytics library into your project and register it with `MobileCore`
+
+   ### Java
+
+   ```java
+   public class MainApp extends Application {
+        private static final String APP_ID = "YOUR_APP_ID";
```
Comment (emdobrin) [inline Documentation/getting-started.md:44]: nit: can we rename to `private final String ENVIRONMENT_FILE_ID = "YOUR_APP_ENVIRONMENT_ID";` for both java/kotlin

File: Documentation/getting-started.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+           MobileCore.registerExtensions(extensions) {
+               Log.d(LOG_TAG, "AEP Mobile SDK is initialized")
+           }
+       }
+   }
+   ```
+
+## Next Steps
+
+Get familiar with the various APIs offered by the AEP SDK by checking out the [Analytics API reference](./api-reference.md).
```
Comment (emdobrin) [inline Documentation/getting-started.md:82]: offered by the Analytics extension by checking out the API reference

File: code/build.gradle (hunk 9 above)
Comment (emdobrin) [inline code/build.gradle:24]: we can keep this one for testing and remove jitpack for now since not used

File: Documentation/getting-started.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
+# Getting Started with Analytics SDK
```
Comment (cacheung) [inline Documentation/getting-started.md:1]: Can we follow the Audience format:
1. Have a title for Adobe Analytics and with description 
The [Analytics](https://developer.adobe.com/client-sdks/documentation/adobe-analytics/) extension represents the Analytics Adobe Experience Platform SDK that is required for sending mobile interaction data from your app to Adobe Analytics.

2. Move the dependence inside the Add Analytics extension to your app.
3. Add Configuring the Analytics extension in the Data Collection UI

**Comment by cacheung** [comment]
Change the license to 2023
https://github.com/adobe/aepsdk-analytics-android/blob/dev/LICENSE#L189

File: Documentation/getting-started.md (hunk 16 above)
Comment (yangyansong-adbe) [reply Documentation/getting-started.md:1]: Emm ... I just followed the steps we use in the Assurance repo. I think it should be better to update it in another PR.

File: code/analytics/build.gradle (hunk 1 above)
Comment (yangyansong-adbe) [reply code/analytics/build.gradle:70]: I saw that we changed to use 2.+ in another PR, so I will not change it here in this PR. Please check and make the correct change in the newer PR. 
cc @kevinlind @emdobrin

File: Documentation/getting-started.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+   ### Kotlin
+
+   ```kotlin
+   class MyApp : Application() {
+       val ENVIRONMENT_FILE_ID = "YOUR_APP_ENVIRONMENT_ID"
+
+       override fun onCreate() {
+           super.onCreate()
+           MobileCore.setApplication(this)
+           MobileCore.configureWithAppID("ENVIRONMENT_FILE_ID")
```
Comment (emdobrin) [inline Documentation/getting-started.md:71]: configureWithAppID(ENVIRONMENT_FILE_ID)

File: code/testapp/build.gradle (hunk 4 above)
Comment (emdobrin) [reply code/testapp/build.gradle:53]: can you revert this change after merge with dev?

**Review by emdobrin — COMMENTED** [review]
looks great, couple cleanup items

File: Documentation/getting-started.md (hunk 16 above)
Comment (cacheung) [reply Documentation/getting-started.md:1]: yes.

**Review by cacheung — APPROVED** [review]

**Review by emdobrin — APPROVED** [review]

**Pull request merged**